### DISCUSSION TITLE
Increasing verbosity of the CPMAddPackage output

### DISCUSSION THREAD

**Discussion by davidstephensonfocusrite** [cpm-cmake/CPM.cmake#611] [Q&A]
At the moment when I fetch a git repository using `CPMAddPackage` i just get the output...
`CPM: Adding package <repository name>@0 (<git tag>)`

But some repositories are quite big and may have submodules that also need fetching which leaves the console output looking stalled for quite some time. It would be nice if we could increase the verbosity of the output. Is this at all possible under the latest version of CPM?

**Comment by Avus-c** [comment]
Yes, it's possible, but it might not be exactly what you're hoping for ... :D

First of all: Set the cache variable `FETCHCONTENT_QUIET` to `FALSE`.
This will ensure that you get more output from CMake beyond just errors. (CPM is built on top of FetchContent.)

If you're just using the `URL` and `URL_HASH` arguments in your CPMAddPackage call, this will generate output like:

```log
[cmake] -- [download 1% complete]
[cmake] -- [download 2% complete]
...
```

If you are actually cloning a Git repository (using `GITHUB_REPOSITORY` and friends), you also need to add `GIT_PROGRESS TRUE` to the `CPMAddPackage` call. This passes the `--progress` flag to the `git clone` command.

Other than that, I don't think there is anything else CMake allows us to enable.

Keep in mind that both these methods come from CMake's underlying FetchContent module, not CPM itself. We have no control of what information is printed.

Also, be aware that enabling these can result in **very very very** verbose output.

It may be more practical to notify the user with a `message` call that the package download may take 30-40 seconds, than flooding the console with too much information.

Hope this helps :)
